### PR TITLE
Migrate to sphinx

### PR DESCRIPTION
**Pull request by oliel**

### CODE REVIEW

File: .github/workflows/sphinx_doc.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: "Pull Request Docs Check"
+on: 
+- pull_request
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: ammaraskar/sphinx-action@master
```
Comment (mwperina) [inline .github/workflows/sphinx_doc.yaml:10]: Hmm, code in this action hasn't been changed for 2 years. Wouldn't it be better just run sphinx from command line? I mean replace pdoc invocation below with the relevant sphinx command?
https://github.com/oVirt/python-ovirt-engine-sdk4/blob/main/.github/workflows/create_docs.yaml#L53

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+sys.path.insert(0, os.path.abspath('../..'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'ovirtsdk4'
+author = 'Ori Liel, Martin Necas, Ondra Machacek, Juan Hernandez'
+
+# The short X.Y version
+version = ''
```
Comment (tinez) [inline docs/source/conf.py:26]: Shouldn't we fill the version and release dynamically somehow (like by passing these as cmdline options to sphinx)

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+
+# -- Options for Texinfo output ----------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (master_doc, 'ovirtsdk4', 'ovirtsdk4 Documentation',
+     author, 'ovirtsdk4', 'One line description of project.',
```
Comment (tinez) [inline docs/source/conf.py:152]: Maybe we could change 'One line description of project' to something more meaningful

**Review by tinez — COMMENTED** [review]
I don't see any changes to 'create_docs.yml' workflow, should we add the sphinx deps and run sphinx there?

File: .github/workflows/create_docs.yaml [hunk 4]
```diff
@@ -50,7 +50,9 @@ jobs:
           ref: 'gh-pages'
 
       - name: Create python documentation
-        run: pdoc -t ovirt-engine-sdk -o ./html ovirtsdk4
+        run:
```
Comment (mwperina) [inline .github/workflows/create_docs.yaml:53]: You have multiple commands executed so you need to use a pipe:

```
run: |
  cd ovirt-engine-sdk/docs
  make html
```

**Review by mwperina — CHANGES_REQUESTED** [review]

**Review by mwperina — COMMENTED** [review]
Try to change line 10 to:

```
workflow_dispatch:
  branches: [ master,sdk_4.4 ]
```

**Pull request closed without merging**